src/core/view/background/BaseBackgroundPainter.cpp

```diff
@@ -18,29 +18,27 @@ void BaseBackgroundPainter::resetConfig() {
 auto BaseBackgroundPainter::alternativeColor(Color color1, Color color2) const -> Color {
     auto backgroundColor = this->page->getBackgroundColor();
 
-    auto greyscale = [](uint32_t color) {
-        return ((0xff & color) + (0xff & (color >> 8)) + (0xff & (color >> 16))) / 3;
-    };
+    auto greyscale = [](Color const& color) { return (uint32_t(color.r) + color.g + color.b) / 3; };
```

Comment:
I am always unsure, what the explicit casting rules are. So I wanted to make sure, that the expression (uint8_t + uint8_t + uint8_t) does not create an overflow. But, I checked both versions, and they are equal, but it might be implementation or platform defined behav.

So to be sure, I would write this: 

```suggestion
    auto greyscale = [](Color const& color) { return (uint32_t(color.r) + uint32_t(color.g) + uint32_t(color.b)) / 3; };
```

Reply:
Ok, all types with sizeof(type) <= sizeof(int) are promoted to (unsigned) int.